marketplaceItems/appreciate/appResources/appData/resources/js/appreciate_ui.js

```diff
@@ -0,0 +1,113 @@
+/*
+    Appreciate
+    Created by Zach Fox on 2019-01-30
+    Copyright 2019 High Fidelity, Inc.
+
+    Distributed under the Apache License, Version 2.0.
+    See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+*/
+
+/* globals document EventBridge setTimeout */
+
+function appreciateSwitchClicked(checkbox) {
+    EventBridge.emitWebEvent(JSON.stringify({
+        method: "appreciateSwitchClicked",
+        appreciateEnabled: checkbox.checked
+    }));
+    document.getElementById("firstRun").style.display = "none";
+}
+
+function neverWhistleCheckboxClicked(checkbox) {
+    var crosshatch = document.getElementById("crosshatch");
+    if (checkbox.checked) {
+        crosshatch.style.display = "inline-block";
+    } else {
+        crosshatch.style.display = "none";
+    }
+
+    EventBridge.emitWebEvent(JSON.stringify({
+        method: "neverWhistleCheckboxClicked",
+        neverWhistle: checkbox.checked
+    }));
+}
+
+var START_COLOR_MULTIPLIER = 0.2;
+function setEntityColor(jscolor) {
+    var newEntityColor = {
+        "red": jscolor.rgb[0],
+        "green": jscolor.rgb[1],
+        "blue": jscolor.rgb[2]
+    };
+
+    var startColor = {
+        "red": Math.floor(newEntityColor.red * START_COLOR_MULTIPLIER),
+        "green": Math.floor(newEntityColor.green * START_COLOR_MULTIPLIER),
+        "blue": Math.floor(newEntityColor.blue * START_COLOR_MULTIPLIER)
+    };
+
+    var currentIntensityDisplayWidth = document.getElementById("currentIntensityDisplay").offsetWidth;
+    var bgString = "linear-gradient(to right, rgb(" + startColor.red + ", " +
```

Comment:
Consider using template strings for complex strings like these for readability(so wish we had that in the interface scripts)

Reply:
I'm not concerned about this since it is a widely-accepted practice! you are totally right that it's harder to read, though.